### PR TITLE
[Runs Feed] Add support for job backfills

### PR DESCRIPTION
**Pull request by salazarm**
## Summary & Motivation

Add support for job backfills

## How I Tested These Changes

Screenshot 2024-09-24 at 12 07 55 PM

## Changelog

NOCHANGELOG

### CODE REVIEW

**Review by jamiedemaria — COMMENTED** [review]
is it intended that the job-backfill version won't have the bar that gets colored in as partitions progress?

**Comment by salazarm** [comment]
>is it intended that the job-backfill version won't have the bar that gets colored in as partitions progress?

Ah, let me add that

File: js_modules/dagster-ui/packages/ui-core/src/instance/backfill/BackfillOpJobPartitionsTable.tsx [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+  const repoAddress = backfill.partitionSet
+    ? buildRepoAddress(
+        backfill.partitionSet.repositoryOrigin.repositoryName,
+        backfill.partitionSet.repositoryOrigin.repositoryLocationName,
+      )
+    : null;
+
+  const inProgress = useMemo(
+    () => results.filter((result) => inProgressStatuses.has(result.runStatus as any)),
```
Comment (hellendag) [inline js_modules/dagster-ui/packages/ui-core/src/instance/backfill/BackfillOpJobPartitionsTable.tsx:26]: Is there a better option than `any` on these?

File: js_modules/dagster-ui/packages/ui-core/src/instance/backfill/BackfillOpJobPartitionsTable.tsx [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    () => results.filter((result) => failedStatuses.has(result.runStatus as any)),
+    [results],
+  );
+
+  if (!backfill.partitionStatuses) {
+    return (
+      <Box margin={48}>
+        <NonIdealState
+          title="Partition statuses unavailable"
+          description="Dagster was unable to load per-partition statuses. This may occur if the backfilled assets or jobs no longer exist in your loaded code locations."
```
Comment (hellendag) [inline js_modules/dagster-ui/packages/ui-core/src/instance/backfill/BackfillOpJobPartitionsTable.tsx:43]: I think we don't normally refer to Dagster in the third person. Maybe just "Unable to load per-partition statuses"?

**Review by hellendag — APPROVED** [review]

File: js_modules/dagster-ui/packages/ui-core/src/instance/backfill/BackfillOpJobPartitionsTable.tsx (hunk 1 above)
Comment (salazarm) [reply js_modules/dagster-ui/packages/ui-core/src/instance/backfill/BackfillOpJobPartitionsTable.tsx:26]: the problem is that `runStatus` could be `null` and `inProgressStatuses` doesn't accept null :(

**Pull request merged**